portfolio/src/main/webapp/script.js

```diff
@@ -30,32 +30,57 @@ function addRandomFact() {
 /**
  * Controls slideshow image display on the page
  */
-var slideIndices = [0, 0, 0]; // array of slide indices for each slide deck
-for(var i = 0; i < slideIndices.length; i++) {
-    showSlides(i, slideIndices[i]);
+const decks = { // Enumerate slide decks
+  DOG_DECK: 0,
+  PLACE_DECK: 1,
+  FOOD_DECK: 2,
+};
+
+const SLIDES = [["/images/dog/dog0.jpg", "/images/dog/dog1.jpg", "/images/dog/dog2.jpg"],
+  ["/images/place/place0.jpg", "/images/place/place1.jpg", "/images/place/place2.jpg"],
+  ["/images/food/food0.jpg", "/images/food/food1.jpg", "/images/food/food2.jpg"]]
+
+let slideIndices = [0, 0, 0]; // array of slide indices for each slide deck
+
+function changeSlide(deckNum, direction) {
+  const deckSize = SLIDES[deckNum].length;
+  let slideIndex = slideIndices[deckNum];
+  slideIndex += direction;
+  if(slideIndex === deckSize) { // wrap around
+      slideIndex = 0;
+  }
+  else if(slideIndex === -1) {
+      slideIndex = deckSize - 1;
+  }
```

Comment:
Why did you change from modular math? I think what you had before was cleaner. If you have a reason, then you don't have to switch back.

Reply:
I realized that going in the backwards direction would go 2, 1, 0, 1 (because it is modded and then the absolute value is taken) so I would have to take the absolute value at a different place in the code for it to work which did not seem legible. If I figure out another way to do that math I'll change it back to modular math